scipy/fft/tests/test_real_transforms.py

```diff
@@ -0,0 +1,87 @@
+
+import numpy as np
+from numpy.testing import assert_allclose
+import pytest
+
+from scipy.fft import dct, idct, dctn, idctn, dst, idst, dstn, idstn
+import scipy.fft as fft
+from scipy import fftpack
+
+# scipy.fft wraps the fftpack versions but with normalized inverse transforms.
+# So, the forward transforms and definitions are already thoroughly tested in
+# fftpack/test_real_transforms.py
+
+
+@pytest.mark.parametrize("forward, backward", [(dct, idct), (dst, idst)])
+@pytest.mark.parametrize("type", [1, 2, 3, 4])
+@pytest.mark.parametrize("n", [2, 3, 4, 5, 10, 16])
+@pytest.mark.parametrize("axis", [0, 1])
+@pytest.mark.parametrize("norm", [None, 'ortho'])
+def test_identity_1d(forward, backward, type, n, axis, norm):
+    # Test the identity f^-1(f(x)) == x
+    x = np.random.rand(n, n)
+
+    y = forward(x, type, axis=axis, norm=norm)
+    z = backward(y, type, axis=axis, norm=norm)
+    assert_allclose(z, x)
+
+    pad = [(0, 0)] * 2
+    pad[axis] = (0, 4)
+
+    y2 = np.pad(y, pad, mode='edge')
+    z2 = backward(y2, type, n, axis, norm)
+    assert_allclose(z2, x)
```

Comment:
I may have missed it elsewhere, but are there any tests that verify that the energy a signal before and after the transform is the same when `norm='ortho'`?

i.e. `assert_allclose(np.linalg.norm(x), np.linalg.norm(y))` should pass when `norm='ortho'`

Reply:
Here I am focusing on testing the differences from `scipy.fftpack` and letting `test_fftpack_equivalence` cover the rest. Does that make sense?

Reply:
Yeah, that made sense. I was just wondering whether this property was ever tested in FFTPack either? That is not strictly an issue for this PR, though, so I approved it as-is.